From 7475daff46de59a11f13711e8d6877fc3e34423d Mon Sep 17 00:00:00 2001
From: samaga Prasanna Krishna <samaga.krishna@intel.com>
Date: Wed, 5 Sep 2012 00:10:18 +0530
Subject: [PATCH 23/49] [PORT FROM MAIN]Audio-MW: system core codec offload
 flag

BZ: 46225

This patch will add codec offload flag to be used in AudioFlinger.

Change-Id: I74c76c15a63fba2710ae4a06f88575549ec5f486
oldChange-Id: I0ff1621f510a32777ca32227f0ca42c4d32b10d6
Signed-off-by: samaga Prasanna Krishna <samaga.krishna@intel.com>
Reviewed-on: http://android.intel.com:8080/78367
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 include/system/audio.h |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/include/system/audio.h b/include/system/audio.h
index cfe8913..cdad428 100644
--- a/include/system/audio.h
+++ b/include/system/audio.h
@@ -371,6 +371,10 @@ enum {
                                AUDIO_DEVICE_IN_FM_RECORD |
                                AUDIO_DEVICE_IN_DEFAULT),
     AUDIO_DEVICE_IN_ALL_SCO = AUDIO_DEVICE_IN_BLUETOOTH_SCO_HEADSET,
+    //devices not supported for codec offload
+    AUDIO_DEVICE_OUT_NON_OFFLOAD = (AUDIO_DEVICE_OUT_ALL & ~(AUDIO_DEVICE_OUT_SPEAKER |
+                                 AUDIO_DEVICE_OUT_WIRED_HEADSET |
+                                 AUDIO_DEVICE_OUT_WIRED_HEADPHONE)),
 };
 
 typedef uint32_t audio_devices_t;
@@ -395,7 +399,8 @@ typedef enum {
                                         // controls related to voice calls.
     AUDIO_OUTPUT_FLAG_FAST = 0x4,       // output supports "fast tracks",
                                         // defined elsewhere
-    AUDIO_OUTPUT_FLAG_DEEP_BUFFER = 0x8 // use deep audio buffers
+    AUDIO_OUTPUT_FLAG_DEEP_BUFFER = 0x8, // use deep audio buffers
+    AUDIO_OUTPUT_FLAG_COMPRESS_OFFLOAD = 0x10 // Use compress offload in DOT
 } audio_output_flags_t;
 
 static inline bool audio_is_output_device(audio_devices_t device)
-- 
1.7.9

